### PR TITLE
fix dashboard page crashes

### PR DESCRIPTION
**Pull request by olexii4**
Signed-off-by: Oleksii Orel <oorel@codenvy.com>

### What does this PR do?
Fix dashboard page crashes.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/4107
https://github.com/eclipse/che/issues/4120
https://github.com/eclipse/che/issues/4121
#4136

#### Changelog
Fix page crashes in dashboard when choosing stack types and during project deletion.

### CODE REVIEW

**Comment by riuvshin** [comment]
@olexii4 does this PR also fixes this issue https://github.com/eclipse/che/issues/4136 ?

File: dashboard/src/app/workspaces/workspaces-config.ts [hunk 1]
```diff
@@ -175,8 +175,8 @@ export class WorkspacesConfig {
         controller: 'ListWorkspacesCtrl',
         controllerAs: 'listWorkspacesCtrl'
       })
-      .accessWhen('/workspace/:namespace*/:workspaceName', locationProvider)
-      .accessWhen('/workspace/:namespace*/:workspaceName/:page', locationProvider)
+      .accessWhen('/workspace/:namespace/:workspaceName', locationProvider)
```
Comment (ashumilova) [inline dashboard/src/app/workspaces/workspaces-config.ts]: looks like merge mistake?

File: dashboard/src/app/workspaces/workspace-details/workspace-projects/workspace-details-projects.controller.ts [hunk 2]
```diff
@@ -57,7 +57,7 @@ export class WorkspaceDetailsProjectsCtrl {
 
     this.namespace = $route.current.params.namespace;
     this.workspaceName = $route.current.params.workspaceName;
-    this.workspaceKey = this.namespace + '/' + this.workspaceName;
+    this.workspaceKey = this.namespace + ':' + this.workspaceName;
```
Comment (ashumilova) [inline dashboard/src/app/workspaces/workspace-details/workspace-projects/workspace-details-projects.controller.ts]: looks like merge mistake.

**Review by bmicklea — APPROVED** [review]

**Comment by olexii4** [comment]
QA tests for dashboard are OK

File: dashboard/src/app/workspaces/workspaces-config.ts (hunk 1 above)
Comment (olexii4) [reply dashboard/src/app/workspaces/workspaces-config.ts]: fixed

File: dashboard/src/app/workspaces/workspace-details/workspace-projects/workspace-details-projects.controller.ts (hunk 2 above)
Comment (olexii4) [reply dashboard/src/app/workspaces/workspace-details/workspace-projects/workspace-details-projects.controller.ts]: fixed

**Review by benoitf — APPROVED** [review]

**Review by akurinnoy — APPROVED** [review]

**Comment by JamesDrummond** [comment]
Is this pending merge status? Please add status label.

**Review by ashumilova — APPROVED** [review]

**Pull request merged**